client/src/utils.js

```diff
@@ -0,0 +1,61 @@
+export const formatDateforAPI = (currDate) => {
+  // getMonth returns 0-11
+  const currMonthStr = String(currDate.getMonth() + 1).padStart(2, "0");
+  // getDate returns 1-31
+  const currDayStr = String(currDate.getDate()).padStart(2, "0");
+  return currDate.getFullYear().toString() + currMonthStr + currDayStr;
+};
+
+export const buildDateRangeRoute = (startDate, endDate) => {
+  return `/api/tasks?start=${formatDateforAPI(
+    startDate
+  )}&end=${formatDateforAPI(endDate)}`;
+};
+
+const getDateTime = (taskDateISOString) => {
+  const taskDate = new Date(taskDateISOString);
+
+  return [
+    taskDate.toLocaleDateString(),
+    taskDate.toTimeString().substring(0, 5),
+  ];
+};
+
+export const convertTaskData = (fetchedTaskData) => {
+  // TODO : get task location from fetched tasks
+  return fetchedTaskData.map(function (task) {
+    const [taskDate, startTime] = getDateTime(task.startDate);
+    const endTime = getDateTime(task.endDate)[1];
+    return {
+      title: task.title,
+      location: "task location to be added",
```

Comment:
I added the location as a required field to the backend. You can now use `task.location` and it should work.

Example of it working below:
![image](https://user-images.githubusercontent.com/24628243/163769661-d361e983-c879-4add-a47a-66b21447e68a.png)

Reply:
Implemented `task.location` field.